hw/mcu/stm/stm32_common/src/hal_timer.c

```diff
@@ -308,26 +409,80 @@ stm32f7_hw_setdown(TIM_TypeDef *regs)
 int
 hal_timer_init(int num, void *cfg)
 {
-    struct stm32f7_hal_tmr *tmr;
+    struct stm32_hal_tmr *tmr;
+    TIM_TypeDef *regs;
 
-    if (num >= STM32F7_HAL_TIMER_MAX || !(tmr = stm32f7_tmr_devs[num]) ||
+    if (num >= STM32_HAL_TIMER_MAX || !(tmr = stm32_tmr_devs[num]) ||
         (cfg == NULL)) {
         return -1;
     }
 
-    tmr->sht_regs = (TIM_TypeDef *)cfg;
+    regs = tmr->sht_regs = (TIM_TypeDef *)cfg;
 
-    if (!IS_TIM_CC1_INSTANCE(tmr->sht_regs)) {
+    if (!IS_TIM_CC1_INSTANCE(regs)) {
         return -1;
     }
 
-    stm32f7_hw_setup(num, tmr->sht_regs);
+    stm32_hw_setup(num, regs);
 
     /*
-     * Stop the timers at debugger. XXX Which TIM?
+     * Stop the timers at debugger.
      */
-    DBGMCU->APB1FZ |= 0x1ff; /* TIM2 - TIM7, TIM12-TIM14 */
-    DBGMCU->APB2FZ |= 0x70003; /* TIM1, TIM8-TIM11 */
+#ifdef TIM1
```

Comment:
Opinion: this is a lot of code space being used for very little gain.

Reply:
What bothers you here is the amount of code (size of .c file) or the amount of generated code (size of .elf/.bin file)?

Reply:
In any case I don't see any problem in just freezing them all, but I'll just use the "macro" names instead.